### PR TITLE
Add NanoVGUtils and VGFontRendering modules

### PR DESCRIPTION
**Pull request by JochenJankowai**
TensorVisBase depends on NanoVGUtils and VGFontRendering so it doesn't make sense to keep them in research-dev.

### CODE REVIEW

**Comment by r-englund** [comment]
@petersteneteg how to trigger the auto format thingy? We added the tag and then manually issued a new build on jenkins?

**Comment by petersteneteg** [comment]
@r-englund yes

**Comment by r-englund** [comment]
> 
> 
> @r-englund yes

We started the wrong build, started the branch when one need to start the PR build

**Comment by JochenJankowai** [comment]
Hello? @martinfalk @petersteneteg @r-englund ?

**Comment by petersteneteg** [comment]
Why not use our regular font rendering?

File: misc/nanovgutils/include/inviwo/nanovgutils/processors/nanovgpickingtestprocessor.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#include <inviwo/core/properties/ordinalproperty.h>
+#include <inviwo/core/ports/imageport.h>
+#include <inviwo/nanovgutils/nanovgutils.h>
+#include <inviwo/core/properties/eventproperty.h>
+
+namespace inviwo {
+
+/** \docpage{org.inviwo.NanoVGPickingTestProcessor, Nano VGPicking Test Processor}
+ * ![](org.inviwo.NanoVGPickingTestProcessor.png?classIdentifier=org.inviwo.NanoVGPickingTestProcessor)
+ * Explanation of how to use the processor.
```
Comment (r-englund) [inline misc/nanovgutils/include/inviwo/nanovgutils/processors/nanovgpickingtestprocessor.h]: Write this

File: misc/nanovgutils/include/inviwo/nanovgutils/processors/nanovgpickingtestprocessor.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * ### Properties
+ *   * __<Prop1>__ <description>.
+ *   * __<Prop2>__ <description>
+ */
+
+/**
+ * \class NanoVGPickingTestProcessor
+ * \brief VERY_BRIEFLY_DESCRIBE_THE_PROCESSOR
+ * DESCRIBE_THE_PROCESSOR_FROM_A_DEVELOPER_PERSPECTIVE
```
Comment (r-englund) [inline misc/nanovgutils/include/inviwo/nanovgutils/processors/nanovgpickingtestprocessor.h]: Can be removed

File: misc/nanovgutils/include/inviwo/nanovgutils/nanovgutils.h [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    std::vector<vec2> getQuadraticBezierCurveControlPoints(
+        const std::vector<vec2> &startAndEndPoints, const std::vector<vec2> &intermediatePoints);
+
+    // Get cubic Bezier Spline Control Points.
+    // https://www.codeproject.com/Articles/31859/Draw-a-Smooth-Curve-through-a-Set-of-2D-Points-wit
+    std::pair<std::vector<vec2>, std::vector<vec2>> getCubicBezierCurveControlPoints(
+        const std::vector<vec2> &knots);
+
+    // Solves a tridiagonal system for one of coordinates (x or y) of first Bezier control points.
+    std::vector<float> GetFirstControlPoints(const std::vector<float> &rhs);
```
Comment (r-englund) [inline misc/nanovgutils/include/inviwo/nanovgutils/nanovgutils.h]: dont use capital letter on first character

**Comment by JochenJankowai** [comment]
> 
> 
> Why not use our regular font rendering?

Back in the day, our fontrendering wasn't very good so I used NanoVG instead. I'll have a look at the 'new' font rendering, maybe the NanoVG version can go.

**Comment by JochenJankowai** [comment]
> 
> 
>     * What is the difference between the added ext/nanovg and https://github.com/memononen/nanovg, would be great if we could use a submodule instead

When I created this, I wasn't able to compile NanoVG into a lib, hence I exported everything manually. I could give it another try, maybe I have learnt something within the last two years.

File: misc/nanovgutils/src/processors/nanovgpickingtestprocessor.cpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                     MouseButton::None, MouseState::Move)
+    , nvgContext_(
+          InviwoApplication::getPtr()->getModuleByType<NanoVGUtilsModule>()->getNanoVGContext()) {
+
+    addPort(outport_);
+
+    addProperty(eventPorperty_);
+}
+
+void NanoVGPickingTestProcessor::process() {
```
[the remark this answers is not in the document]
Comment (JochenJankowai) [reply misc/nanovgutils/src/processors/nanovgpickingtestprocessor.cpp]: This is good since the picking was never merged into nanovg.

**Comment by r-englund** [comment]
> Reason 1: When I created this, I wasn't able to compile NanoVG into a lib, hence I exported everything manually. I could give it another try, maybe I have learnt something within the last two years.

I see, the NanoVG does not export any of its method / classes to dll which mean they only support static linking. As I see it, there is two way of addressing this, A: add the exporting ourself (as you have done) orB: force static linking (`add_library(target STATIC sroucefiles)`) 

> Reason 2: The original repo does not support picking. There was a PR that was never merged. This version of NanoVG is based on the fork.

Is the picking in the PR good, fast? See the comment I made on the Picking example processor on how picking can be achieved with the inviwo picking mapper

**Comment by JochenJankowai** [comment]
> I see, the NanoVG does not export any of its method / classes to dll which mean they only support static linking. As I see it, there is two way of addressing this, A: add the exporting ourself (as you have done) orB: force static linking (`add_library(target STATIC sroucefiles)`)

A third option is to just use nanovg as it was intended (as mentioned several times by the author is some of his issues) and simply include it, no libs and all.

**Pull request closed without merging**